src/distilbert/distilbert_model.rs

```diff
@@ -682,30 +682,59 @@ impl DistilBertForTokenClassification {
     /// let config = DistilBertConfig::from_file(config_path);
     /// let distil_bert = DistilBertForTokenClassification::new(&p.root() / "distilbert", &config);
     /// ```
-    pub fn new<'p, P>(p: P, config: &DistilBertConfig) -> DistilBertForTokenClassification
-    where
-        P: Borrow<nn::Path<'p>>,
+    pub fn try_new<'p, P>(p: P, config: &DistilBertConfig) -> Result<DistilBertForTokenClassification, RustBertError>
+        where
+            P: Borrow<nn::Path<'p>>,
     {
         let p = p.borrow();
 
         let distil_bert_model = DistilBertModel::new(p, config);
 
-        let num_labels = config
-            .id2label
-            .as_ref()
-            .expect("id2label must be provided for classifiers")
-            .len() as i64;
-
-        let classifier = nn::linear(p / "classifier", config.dim, num_labels, Default::default());
-        let dropout = Dropout::new(config.seq_classif_dropout);
-
-        DistilBertForTokenClassification {
-            distil_bert_model,
-            classifier,
-            dropout,
+        if let Some(id2label) = &config.id2label {
```

Comment:
Could raising the error be closer to the problematic field using 
```rust
        let num_labels = config
            .id2label
            .as_ref()
            .ok_or_else(|| RustBertError::InvalidConfigurationError("id2label must be provided in the configuration for sequence classification".to_string()))?
            .len() as i64;
```
This also makes this code block more generic and reusable across the crate

Reply:
Thank's fror the hint - didn't think of that +1